python/pyspark/sql/dataframe.py

```diff
@@ -470,7 +470,7 @@ def sampleBy(self, col, fractions, seed=None):
         +---+-----+
 
         """
-        if not isinstance(col, str):
+        if not isinstance(col, basestring):
```

Comment:
According to https://github.com/apache/spark/commit/f958f27e2056f9e380373c2807d8bb5977ecf269, it seems to be possible to use Non-ascii characters in column name.
I think there are use cases which want to use non-ascii character in column name.